pylint_odoo/checkers/no_modules.py

```diff
@@ -408,6 +414,11 @@ def visit_functiondef(self, node):
                 self.add_message('method-required-super',
                                  node=node, args=(node.name))
 
+        if self.linter.is_message_enabled('old-api7-method-defined'):
+            first_args = [arg.name for arg in node.args.args][:3]
+            if first_args == ['self', 'cr', 'uid']:
```

Comment:
I'm thinking about other methods that passes these 2 arguments are for sure in old API. Methods outside a class don't have `self`. You should also look for `cursor` and `user`, that are other aliases. And finally, I think we can check also if there are instructions like `self.pool`.